### PR TITLE
10.0 FIX automatic validation of pickings if the domain returns more than one.

### PR DESCRIPTION
**Pull request by jaredkipe**
If you have your automatic workflow setup to validate pickings (e.g. import some retail/POS orders), the current method will give the following error.

```
ERROR test10113 odoo.addons.sale_automatic_workflow.models.automatic_workflow_job: Error during an automatic workflow action.
 Traceback (most recent call last):
   File "/opt/odoo/addons/sale_automatic_workflow/models/automatic_workflow_job.py", line 23, in savepoint
     yield
   File "/opt/odoo/addons/sale_automatic_workflow/models/automatic_workflow_job.py", line 83, in _validate_pickings
     pickings.validate_picking()
   File "/opt/odoo/addons/sale_automatic_workflow/models/stock_picking.py", line 21, in validate_picking
     self.do_transfer()
   File "/opt/odoo/addons/connector_ecommerce/models/stock.py", line 34, in do_transfer
     result = super(StockPicking, self_context).do_transfer()
   File "/opt/odoo/odoo/addons/delivery/models/stock_picking.py", line 113, in do_transfer
     self.ensure_one()
   File "/usr/local/lib/python2.7/site-packages/odoo/models.py", line 4823, in ensure_one
     raise ValueError("Expected singleton: %s" % self)
 ValueError: Expected singleton: stock.picking(17626, 17627,...)
```

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please change module version number.

**Comment by jaredkipe** [comment]
Done.  May I ask why?  This change doesn't modify any fields or require an 'upgrade'.

**Comment by pedrobaeza** [comment]
Check https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers for the guidelines. 

The goal is to distinguish through numbering the version that is working from the version is not.

**Comment by jaredkipe** [comment]
Did I get it right? (usually just a server restart is required)

**Comment by pedrobaeza** [comment]
Yes, thanks!

**Review by pedrobaeza — APPROVED** [review]

**Comment by jaredkipe** [comment]
@pedrobaeza Anything more required to get his merged?

**Comment by pedrobaeza** [comment]
Yes, you need at least 2 approvals and 5 working days or 3 approvals: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#please-respect-a-few-basic-rules

You can review other PRs and ask in exchange to review yours.

**Comment by jaredkipe** [comment]
@guewen FWIW this does affect the Magento Connector as a dependency ;)

**Review by LoisRForgeFlow — APPROVED** [review]

**Review by atchuthan — CHANGES_REQUESTED** [review]
Suggestion
On line https://github.com/OCA/sale-workflow/blob/10.0/sale_automatic_workflow/models/automatic_workflow_job.py#L81

Instead of 
if pickings:
  with savepoint(self.env.cr):
  pickings.validate_picking()

we could replace with 
for picking in pickings:
  with savepoint(self.env.cr):
  picking.validate_picking()

As this change will automatically save the changes done for each picking instead of committing changes completely in validate_picking()

**Comment by jaredkipe** [comment]
@atchuthan 
That is fair, and makes it more similar to other automatic workflow jobs.

I've left the other commit as the method is decorated `@api.multi` and will now be suitable for using as a server action on its own (e.g. on a list view).

**Review by atchuthan — APPROVED** [review]
sure :+1:

**Pull request merged**
